src/pages/index.js

```diff
@@ -34,69 +38,151 @@ const modalImageElement = imagePreviewModal.querySelector(".modal__image-preview
+  })
+  .catch((err) => {
+    console.error(err); 
+  });
+  
+const profileUserInfo = new UserInfo(".profile__title", ".profile__description", ".profile__image");    
+  
+api
+  .loadUserInfo()
+  .then((res) => {
+    profileUserInfo.getUserInfo();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-// Create instances of the classes
-const cardSection = new Section({
-  items: initialCards,
-  renderer: (cardData) => {
-    const cardEl = renderCard(cardData);
-    cardSection.addItem(cardEl);
-  },
-},  
-".cards__list"
-);
 
-cardSection.renderItems();
 
-const profileEditFormPopup = new PopupWithForm("#profile-edit-modal", handleProfileEditSubmit);
-profileEditFormPopup.setEventListeners();
 
-const editFormElement = profileEditModal.querySelector(".modal__form_edit");
-const editFormValidator = new FormValidator(validationSettings, editFormElement);
-editFormValidator.enableValidation();
+const editFormElement = profileEditModal.querySelector(".modal__form_edit");  
+const editFormValidator = new FormValidator(validationSettings, editFormElement);  
+editFormValidator.enableValidation();  
 
-const profileAddCardPopup = new PopupWithForm("#profile-add-card-modal", handleProfileAddCardSubmit);
-profileAddCardPopup.setEventListeners();
+const avatarFormElement = avatarEditModal.querySelector(".modal__form_avatar");  
+const avatarFormValidator = new FormValidator(validationSettings, avatarFormElement);  
+avatarFormValidator.enableValidation();
 
-const addFormElement = profileAddCardModal.querySelector(".modal__form_add");
-const addFormValidator = new FormValidator(validationSettings, addFormElement);
-addFormValidator.enableValidation();
+const addFormElement = profileAddCardModal.querySelector(".modal__form_add");  
+const addFormValidator = new FormValidator(validationSettings, addFormElement);  
+addFormValidator.enableValidation();  
 
-const profileUserInfo = new UserInfo(".profile__title", ".profile__description");
 
-const popupWithImage = new PopupWithImage("#image-preview-modal");
+const popupWithImage = new PopupWithImage("#image-preview-modal");  
 popupWithImage.setEventListeners();
 
-// Initialize all my instances
+function handlePreviewPicture(name, link) {
+  popupWithImage.open(name, link);
+}
 
 
-// all the rest
 function renderCard(cardData) {
-  return new Card(cardData, "#card-template", () => {
-    popupWithImage.open(cardData.name, cardData.link)
-  })
-  .generateCard();
+  const card = new Card(
+    cardData, 
+    "#card-template",
+    handlePreviewPicture,
+    handleDeleteClick,
+    handleLikeClick,
+  );
+  return card.generateCard();
 }
 
+const profileEditFormPopup = new PopupWithForm("#profile-edit-modal", handleProfileEditSubmit);  
+profileEditFormPopup.setEventListeners();
 
 function handleProfileEditSubmit(inputValues) {
-  profileUserInfo.setUserInfo(inputValues.title, inputValues.description);
-  profileEditFormPopup.close();
+  profileEditFormPopup.setLoading(true);

```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```